core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -333,6 +334,9 @@ private void registeringSubscriptionQueryReadSafe(Predicate<?> subscriptionQuery
         return uow.executeWithResult(() -> {
             ResponseType<R> responseType = uow.getMessage().getResponseType();
             Object queryResponse = new DefaultInterceptorChain<>(uow, handlerInterceptors, handler).proceed();
+            if (queryResponse instanceof Future) {
```

Comment:
To be fair with you, I expected an adjustment in the `ResponseType#convert(Object)` function, rather than an adjustment in a specific `QueryBus` implementation.

My hunch is that would mean two adjustments, being:

1. Adjusting the `default convert` function in the `ResponeType` interface to perform this check.
2. Adjusting the `MultipleInstanceResponseType#convert(Object)` function to perform this check.

That way the `SimpleQueryBus` impl., and any future `QueryBusy` implementation, do not have to care about knowing whether a `Future` is returned or not.

What's your take on this suggestions @m1l4n54v1c (would like to hear @abuijze his thoughts on this as well).

Reply:
Crossed my mind, but felt like too much of responsibility for a conversion. I definitely like the idea of reducing the work for `QueryBus` implementations. Don't have any (other) argument against moving responsibility to `ResponseType`.

Reply:
Just discussed this with @abuijze, he had a different suggestion in mind. 
He's currently on an airplane if I'm correct though, so we'll have to wait 'till he has landed.

Reply:
One more thing came to my mind - if we are going to have `AsyncQueryBus` it wouldn't be a good idea to block the call. So, how we are going to resolve that future is dependent on `QueryBus` implementation. For `SimpleQueryBus` it does not matter anyway where this future is resolved.

Reply:
Of course, query side will not be able to do async things. Will try to get rid of blocking here.